apps/sms/services/js/activity/activity_shim.js

```diff
@@ -78,31 +42,83 @@
    */
   var ActivityShim = {
     /**
+     * Reference to currently active activity request.
+     * @type {ActivityRequestHandler}
+     */
+    [priv.request]: null,
+
+    /**
      * Initialized activity service bridge.
-     * @type {number} appInstanceId Unique identifier of the app instance that
-     * is used to establish 1-to-1 only connection between this service and
-     * corresponding client hosted in the same app instance.
      */
-    init(appInstanceId) {
-      if (!appInstanceId) {
-        throw new Error('App instance id is not specified!');
+    init() {
+      this.initService();
+
+      // We use setTimeout here to allow activity client to do connection
+      // handshake with the service first, so that it will be able to receive
+      // event broadcasted by service.
+      setTimeout(() => {
```

Comment:
I'm referring to this line [1], I assume it's public one, but yeah let's confirm :)

Hey @wilsonpage, @gmarty is "connect" event on service instance public and can be used outside of threads library?

[1] https://github.com/gaia-components/threads/blob/master/lib/service.js#L107